src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -1204,7 +1213,8 @@ DeferErrorIfUnsupportedTableCombination(Query *queryTree)
 		 */
 		if (rangeTableEntry->rtekind == RTE_RELATION ||
 			rangeTableEntry->rtekind == RTE_SUBQUERY ||
-			rangeTableEntry->rtekind == RTE_RESULT)
+			rangeTableEntry->rtekind == RTE_RESULT ||
+			IsJsonTableRTE(rangeTableEntry))
```

Comment:
Because as of today we don't have any conditions that could prevent us from pushing down a json table expr (e.g., we cannot pushdown a VALUES expr when it contains mutable function call as we check below), we treat json table exprs as if they're simple set returning things, like relations and subqueries.

Reply:
(like a self-note for me to remember the reasoning here if I again need to look into this in the future)